### ISSUE TITLE
<BryntumGantt> React wrapper does not pass on configuration for all features

### ISSUE THREAD

**Issue by jashworth** [bryntum/support#2751]
Bryntum Gantt / 4.1.1

The <BryntumGantt> React wrapper does not pass on configuration for all features.

```typescript
import * as React from "react";
import { BryntumGantt } from "@bryntum/gantt-react";
import { Gantt, GanttConfig } from "@bryntum/gantt/gantt.umd.js";
import "@bryntum/gantt/gantt.stockholm.css";
import "./App.css";

interface BryntumGanttRef {
  instance: Gantt;
}

const ganttConfig: Partial<GanttConfig & { [key: string]: unknown }> = {
  dependenciesFeature: false, // ok

  // ignored!
  taskDragFeature: false,
  taskDragCreateFeature: false,
  taskResizeFeature: false,
  // there may be others...it's hard to tell...
};

function App() {
  const ganttRef = React.useRef<BryntumGanttRef>(null!);

  React.useEffect(() => {
    const features: any = ganttRef.current.instance.features;

    [
      "dependencies", // false, as expected

      // not false! (still a reference)
      "taskDrag",
      "taskDragCreate",
      "taskResize",
    ].forEach((feature) => {
      console.log(feature, features[feature]);
    });
  }, []);

  return <BryntumGantt ref={ganttRef} {...ganttConfig} />;
}

export default App;

```

This would be evident during development given fuller fidelity type information.

The list of featureNames in the wrapper

```javscript
// BryntumGantt.js
    featureNames = [
        'baselinesFeature',
        'cellEditFeature',
        'cellMenuFeature',
```

is prone to get (or remain) out of sync with the list of features supported by @bryntum/gantt.

Perhaps the list of features should be exported from @bryntum/gantt, or a type relationship established between the two.

edited: to add version info

**Comment by SergeyMaltsev** [comment]
Hi!

Features in react Wrapper are suffixed with `Feature` and act as both configs and properties for `BryntumGanttComponent`. 
They are mapped to the corresponding API features of the instance.
So those feature names from wrapper are not the part of API Types. 
It is only wrapper config.

This was made for wrapper to make each `FeatureName` wrapper property reactive.

Use may use a simple `object` instead of typed `Partial<GanttConfig>` to be passed as a destructed config.

```
const ganttConfig = {
   ...
   taskDragFeature: true
}

...

return <BryntumGantt 
   ref = { ganttRef } 
   { ...ganttConfig } 
/>;
```

OR use a separate Wrapper config for feature

```
const ganttConfig: Partial<GanttConfig> = {
...
};

<BryntumGantt
    ref = { ganttRef }
    { ...ganttConfig }
    taskDragFeature = { true }
/>
```

**Comment by jashworth** [comment]
Did you try the code I provided?

The three listed features are not passed on by the React wrapper.

Try setting any of them to false, using either of your methods, and observe that it doesn't disable the corresponding feature.

![image](https://user-images.githubusercontent.com/4151375/116076828-a8a5c300-a68c-11eb-9e04-77f19b387e4f.png)

The only way to do it is to use

```typescript
features: {
  taskDrag: false,
  taskDragCreate: false,
  taskResize: false,
}
```

which produces

![image](https://user-images.githubusercontent.com/4151375/116077215-236ede00-a68d-11eb-9f2d-b565c717bbd3.png)

but that method is deprecated

![image](https://user-images.githubusercontent.com/4151375/116076397-14d3f700-a68c-11eb-9c3b-c0b1641dcba8.png)

**Comment by SergeyMaltsev** [comment]
Yes. This is all correct.

You should use feature configs on react component as I mentioned above.
Not possible to use in `GanttConfig`

Feature should be disabled by 
```
<BryntumGantt
    ref = { ganttRef }
    { ...ganttConfig }
    taskDragFeature = { 
       disabled : true 
    }
/>
```

Please check docs
https://www.bryntum.com/docs/gantt/#Gantt/feature/TaskDrag#config-disabled

**Comment by jashworth** [comment]
Your code

```typescript
<BryntumGantt
    ref = { ganttRef }
    { ...ganttConfig }
    taskDragFeature = { 
       disabled : true 
    }
/>
```

does not compile.

I have already tried using feature configuration on the React component.

In particular, I've tried

```typescript
    <BryntumGantt
      ref={ganttRef}
      {...ganttConfig}
      taskDragFeature={{ disabled: true }}
      taskDragCreateFeature={{ disabled: true }}
      taskResizeFeature={{ disabled: true }}
    />
```

and

```typescript
    <BryntumGantt
      ref={ganttRef}
      {...ganttConfig}
      taskDragFeature={false}
      taskDragCreateFeature={false}
      taskResizeFeature={false}
    />
```

and tasks can still be dragged and resized.

**Comment by SergeyMaltsev** [comment]
Hi!

Thanks for the report. 
We had a problem with missing those configs for Gantt.
It is fixed.

Could you please try with the recently released 4.1.2 version?

```    <BryntumGantt
      ref={ganttRef}
      {...ganttConfig}
      taskDragFeature={{ disabled: true }}
      taskDragCreateFeature={{ disabled: true }}
      taskResizeFeature={{ disabled: true }}
    />
```

**Comment by jashworth** [comment]
The wrapper is now passing on the configuration correctly.
